crates/project_model/src/build_data.rs

```diff
@@ -172,9 +173,17 @@ impl WorkspaceBuildData {
         cmd.stdout(Stdio::piped()).stderr(Stdio::piped()).stdin(Stdio::null());
 
         let mut child = cmd.spawn().map(JodChild)?;
+
         let child_stdout = child.stdout.take().unwrap();
         let stdout = BufReader::new(child_stdout);
 
+        let mut stderr = BufReader::new(child.stderr.take().unwrap());
+        let mut stderr_vec = Vec::new();
+        thread::scope(|s| {
+            s.spawn(|_| stderr.read_to_end(stderr_vec.as_mut()).unwrap());
```

Comment:
I think the `for` loop needs to be inside of the `crossbeam` scope. Otherwise the scope will join the threads when it's dropped, which means that the code will wait for `stderr` to end (i.e. for `cargo check` to finish). This means we don't process progress messages in the `for` loop before `cargo check` exists.